aquadoggo/src/network/replication/behaviour.rs

```diff
@@ -78,21 +104,49 @@ impl NetworkBehaviour for Behaviour {
     fn on_connection_handler_event(
         &mut self,
         peer: PeerId,
-        _connection_id: ConnectionId,
+        connection_id: ConnectionId,
         handler_event: THandlerOutEvent<Self>,
     ) {
         match handler_event {
             HandlerOutEvent::Message(message) => {
-                self.handle_received_message(&peer, message);
+                self.handle_received_message(&peer, message, connection_id);
             }
         }
     }
 
     fn on_swarm_event(&mut self, event: FromSwarm<Self::ConnectionHandler>) {
         match event {
-            FromSwarm::ConnectionEstablished(_)
-            | FromSwarm::ConnectionClosed(_)
-            | FromSwarm::AddressChange(_)
+            FromSwarm::ConnectionClosed(ConnectionClosed {
```

Comment:
This is where we pass events up to the swarm which will then be caught and passed onto the service bus. The reason I did it here as opposed to directly on the swarm events is because we don't have access to `ConnectionId` there, which we want to pass onto the `ConnectionManager`.

Reply:
This goes some way to solving https://github.com/p2panda/aquadoggo/issues/399 as when a connection suddenly closes it eventually times out and a `ConnectionClosed` event is sent. We may also need to work on the `KeepAlive` logic in the handler, we could give it a `KeepAlive::Until(x)` which might help libp2p tidy up stale connections.